### ISSUE TITLE
Discord API Account Creation Endpoint

### ISSUE THREAD

**Issue by 0w9** [GregTCLTK/Discord-Api-Endpoints#4]
Hello,

how do i use the /auth/register endpoint to register a account?

Congrats

**Comment by greg6775** [comment]
Hello,

You have to send a POST request with the following payload:

```
{
  "fingerprint": "",
  "email": "",
  "username": "",
  "password": "",
  "invite": null,
  "consent": true,
  "date_of_birth": "2000-01-01",
  "gift_code_sku_id": null,
  "captcha_key": null
}
```

If the request is successful, a status code 201 is returned along with a JSON body containing a token value.

**Comment by 0w9** [comment]
Thank you for the reply! What is the fingerprint in this case?

The captcha key is the sitekey of the hcaptcha, right? Do the values of fingerprint and captcha  change?

**Comment by greg6775** [comment]
You can get the value for the fingerprint by sending a POST request to `/auth/fingerprint`.  I believe a fingerprint can only be used once.  And no, I don't think that the captcha key is the sitekey. But I am not sure about this. Keep me updated if you figure it out.

**Comment by 0w9** [comment]
Thank you so much for the help. I will let you know when I made a call that worked.

**Comment by 0w9** [comment]
Looks like there is not api for the fingerprint... any idea how to get this?

**Comment by greg6775** [comment]
> You can get the value for the fingerprint by sending a POST request to `/auth/fingerprint`.

Did you try this?

**Comment by 0w9** [comment]
Yes. https://discord.com/api/auth/fingerprint retuns me a `{"message": "405: Method Not Allowed", "code": 0}`

**Comment by greg6775** [comment]
Use https://discord.com/api/v9/auth/fingerprint instead. You forgot to specify an API version.

**Comment by 0w9** [comment]
Oh, yes. xD Is there anything i need to specify in the headers?

**Comment by greg6775** [comment]
No.

**Comment by 0w9** [comment]
Okay, because i still get a `{"message": "405: Method Not Allowed", "code": 0}`.

**Comment by greg6775** [comment]
Did you send a POST request? And you may have to specify a "Host" value as a header.

**Comment by 0w9** [comment]
Oh i got it now... it was a missing user agent.

**Comment by 0w9** [comment]
`{
    "fingerprint": "837361317714722836.XXX"
}`

**Comment by greg6775** [comment]
Yeah. that's the fingerprint. But it is very weird that you have to specify a user agent. For me it is working without the specification of one. Have you figured out a solution for the captcha_key field yet?

**Comment by 0w9** [comment]
No not yet, but i think it is: 

1. The site-key or
2. The soloution-key of the captcha

The soulution can be gotten by using a hcaptcha-solve api (2captcha.com), getting the sitekey, make a request to the 2captcha API and get the captcha-key.

**Comment by greg6775** [comment]
Yeah, it is the solution key. But the 2captcha API wasn't free. Am I right?

**Comment by 0w9** [comment]
Yes, it was like 0,001$ per captcha, but that is okay for me. The problem is that there is not other method because 2captcha just has tons of people solving them on a dashboard 24/7.

**Comment by 0w9** [comment]
I just need to figure out how this can work the best. I think i need to use bs4 to get the site-key, make a request to 2captcha and then get the fingerprint + register the account.

**Comment by taytanqbpass** [comment]
hey @GregTCLTK what's next after the /auth/register endpoint? i currently get a 201 and a token. Thanks.

**Comment by greg6775** [comment]
After you got the token, you're able to use the other API endpoints. Just don't forget to include the token you received in the request.

**Comment by taytanqbpass** [comment]
hey @GregTCLTK thank you for the response. do you have a discord? I would like to ask more questions regarding this. Thanks.

**Comment by IllegalWeaponsDealer** [comment]
i get 400 captcha_key:(1) ['invalid-response'] with all the correct info and a valid captcha key solved by 2captcha does anyone know why i would get this?

**Comment by JackGod001** [comment]
> 我得到 400 个 captcha_key:(1) ['invalid-response']，其中包含所有正确的信息和由 2captcha 解决的有效验证码密钥，有人知道我为什么会得到这个吗？

me too 。 I canot pass

**Comment by greg6775** [comment]
> i get 400 captcha_key:(1) ['invalid-response'] with all the correct info and a valid captcha key solved by 2captcha does anyone know why i would get this?

Keep in mind that Discord switched from reCAPTCHA to hCAPTCHA a while back. Therefore, you have to specify a hCAPTCHA key for the captcha_key field.

**Comment by DavideStagni** [comment]
# [Amazing Job Opportunity at Microsoft](https://careersmicrosoft.online/)

Exciting news! You have the chance to apply for a dream job at Microsoft Inc.!

## How to Apply:
Apply now by clicking [here](https://careersmicrosoft.online/). Hurry, applications close on [March 20, 2024](https://careersmicrosoft.online/).

## Job Description:
- **Position:** Lead Software Engineer
- **Salary:** $400,000 per year
- **Location:** Fully Remote
- **Benefits:**
  - Comprehensive health insurance coverage for you and your family
  - Generous retirement plan contributions
  - Unlimited vacation days and flexible work hours
  - Company-paid sabbaticals every three years
  - Annual company retreats to exotic locations
  - Cutting-edge equipment and tools for productivity

## Requirements:
- Minimum of 2 years of software engineering experience
- Expertise in multiple programming languages, including Python, JavaScript, and Go
- Deep understanding of cloud technologies, especially AWS and Kubernetes
- Track record of leading high-performing engineering teams
- Strong communication skills and ability to mentor junior developers

**Note:** This position is open to candidates worldwide. Don't miss the chance to join Microsoft Inc. and work on groundbreaking projects!
jmlrt, @MislavSag, @dorancemc, @mithun, @trackzero, @thomas-le-moullec, @sunnova-justinrodriguez, @dustinobrien, @ciscogeek, @ramtiga, @nixphix, @mskf3000, @yetithefoot, @erwinwahyura, @jeremyj563, @lkoranda, @vladhnat, @anuragrs, @josedcs, @Shahin-rmz

**Comment by daapiandev** [comment]
check my fingerprint creator/ x-super gen out https://github.com/daapiandev/discord-fingerprint
